### PR TITLE
feature: enable gesture to change emoji category with horizontal swipe gesture

### PR DESCRIPTION
**Pull request by pierre275**
# Description
This pr adds new props `enableCategoryChangeGesture` which let user to change category with horizontal swipe

#### How to test
Follow provided documentation and check added usage example

### CODE REVIEW

**Review by jakex7 — CHANGES_REQUESTED** [review]
Great feature, but it needs a little tuning to be included. For a better user experience, the category Indicator should be moved during scroll as well.
If you need more details please let me know 😃 

image

File: src/contexts/KeyboardContext.tsx [hunk 1]
```diff
@@ -77,6 +77,8 @@ export type KeyboardProps = {
   styles?: RecursivePartial<Styles>
   enableSearchAnimation?: boolean
   enableCategoryChangeAnimation?: boolean
+
+  enableCategoryChangeGesture?: boolean
```
Comment (oferRounds) [inline src/contexts/KeyboardContext.tsx:81]: missing also from `defaultKeyboardValues` object

**Review by oferRounds — COMMENTED** [review]
Great-needed feature

File: src/contexts/KeyboardContext.tsx (hunk 1 above)
Comment (oferRounds) [reply src/contexts/KeyboardContext.tsx:81]: I see now that `enableCategoryChangeGesture` is also missing from there

**Comment by mateki0** [comment]
> Great feature, but it needs a little tuning to be included. For a better user experience, the category Indicator should be moved during scroll as well. If you need more details please let me know 😃
> 
> image

I've added this feature in the latest commit

File: src/components/EmojiStaticKeyboard.tsx [hunk 2]
```diff
@@ -60,6 +66,16 @@ export const EmojiStaticKeyboard = React.memo(
     }, [activeCategoryIndex, enableCategoryChangeAnimation, shouldAnimateScroll])
 
     const keyExtractor = React.useCallback((item: EmojisByCategory) => item.title, [])
+    const scrollNav = React.useRef(new Animated.Value(0)).current
+
+    const handleScroll = React.useCallback(
+      (el: NativeSyntheticEvent<NativeScrollEvent>) => {
+        const index = el.nativeEvent.contentOffset.x / width
+        scrollNav.setValue(index * CATEGORY_ELEMENT_WIDTH)
```
Comment (jakex7) [inline src/components/EmojiStaticKeyboard.tsx:74]: I've just changed Spring animation to setValue because now animation feel lot more responsive @mateki0

**Comment by jakex7** [comment]
Thanks @pierre275 @mateki0, It's so great that i've enabled it by default 🚀

**Review by jakex7 — APPROVED** [review]

**Pull request merged**